### DISCUSSION TITLE
How to ignore `arch()->preset()->laravel();`?

### DISCUSSION THREAD

**Discussion by laudaikinhdi** [pestphp/pest#1248] [Q&A]
I using `arch()->preset()->laravel();` but I want to ignore this lines [source](https://github.com/pestphp/pest/blob/3.x/src/ArchPresets/Laravel.php#L148-L149)
```php
$this->expectations[] = expect('App\Http')
    ->toOnlyBeUsedIn('App\Http');
```

**Comment by sinnbeck** [comment]
I havent had a chance to try it, but I think you can customize what to ignore

https://pestphp.com/docs/arch-testing#content-modifiers

**Reply by laudaikinhdi** [reply]
Thanks for reply. Can you give me example code?

**Reply by owenvoke** [reply]
It doesn't look like you can exclude only one expectation rule. 🤔

You could use this, but it would ignore all the Laravel preset expectations for that namespace, which I assume you don't want.
```php
arch()
    ->preset()
    ->laravel()
    ->ignoring('App\Http');
```

It would be good to maybe have a way to just target ignoring a single rule, but they don't have any names or identifiers so not sure how that would work without changing the design of presets.

**Reply by sinnbeck** [reply]
I would assume this
```php
arch()
->preset()
->laravel()
->ignoring('App\Http');
```

But you might want to add this manually.
```php
arch()->expect('App\Http\Controllers')
            ->classes()
            ->toHaveSuffix('Controller');

**Comment by inmanturbo** [comment]
> I using `arch()->preset()->laravel();` but I want to ignore this lines [source](https://github.com/pestphp/pest/blob/3.x/src/ArchPresets/Laravel.php#L148-L149)
> 
> ```
> $this->expectations[] = expect('App\Http')
>     ->toOnlyBeUsedIn('App\Http');
> ```

This seems crazy. What about route files, how to add controller actions to routes?

**Reply by laudaikinhdi** [reply]
Because I am applying services and repositories, I placed them outside the Http folder. Sometimes, I need to use FormRequest in the service or repository.

**Reply by inmanturbo** [reply]
Yeah I get that. I meant the preset expectation seems a little crazy, unless I'm mistaken. I mean how to even do routing?
